### PR TITLE
Support for signac 2

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR refactors signac-flow to support signac 2. I am testing against the current HEAD of signac.

**Reviewers and maintainers:** I have not tested this PR against any production-level projects, only the pytests written for signac-flow. I strongly recommend that this branch should be tested with real clusters and production-ready projects before merging.

## Motivation and Context
Addresses part of https://github.com/glotzerlab/signac/issues/831.

## Checklist:
- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [ ] The [package documentation](https://github.com/glotzerlab/signac-flow/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
  - [ ] Before merging, we should read through the docs and make sure we're not missing anything obvious that the signac 2 version change would affect.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

**Note:** I have not updated the changelog yet because I want to merge all other open PRs and make a 0.24 release before merging this PR.

### CODE REVIEW

**Review by b-butler — APPROVED** [review]
Looks good to me.

File: pyproject.toml [hunk 1]
```diff
@@ -30,7 +30,9 @@ classifiers = [
 ]
 dependencies = [
     # The core package.
-    "signac>=1.8.0",
+    # TODO: Test against the released 2.0.0 once available.
+    #"signac>=2.0.0",
+    "signac @ git+https://github.com/glotzerlab/signac@master",
```
Comment (kidrahahjo) [inline pyproject.toml:35]: Created an issue to track this one https://github.com/glotzerlab/signac-flow/issues/711

**Comment by vyasr** [comment]
@b-butler @tcmoore3 could you get a couple of people to test this on some projects (or do so yourselves)? Once that's done I think we can merge, but as @bdice says we should definitely get this tested by someone on a production project.

**Comment by b-butler** [comment]
I will test on one of my projects and report back.

**Comment by bdice** [comment]
> I will test on one of my projects and report back.

@b-butler Can you update this PR with a description of what you (or others) have tested? Aggregation, groups, submission, status, etc?

File: pyproject.toml [hunk 2]
```diff
@@ -30,7 +30,9 @@ classifiers = [
 ]
 dependencies = [
     # The core package.
-    "signac>=1.8.0",
+    # TODO: Test against the released 2.0.0 once available.
+    #"signac>=2.0.0",
```
Comment (bdice) [inline pyproject.toml:34]:
```suggestion
    # "signac>=2.0.0",
```

**Comment by b-butler** [comment]
I tested submission, run, and status. I did not use groups or aggregates for the project I tested in practice.

**Review by b-butler — APPROVED** [review]
I am ready for this to proceed when dashboard is done.

**Comment by bdice** [comment]
@b-butler I would propose the following sequence:
- Get all repos ready for signac 2 (docs, examples are the primarily ones left)
- Release signac 2.0.0
- Merge flow/dashboard PRs
- Release flow/dashboard

I could see the motivation to merge flow/dashboard PRs before releasing signac 2.0.0 but it would require another PR to fix the dependencies once 2.0.0 is released. Thoughts welcome.

**Comment by b-butler** [comment]
@bdice I generally agree. I would like to have all the PRs open, and when signac 2.0 is merged/released, I would want to update the requirements in all the PRs and begin releases where appropriate.

**Pull request merged**